rafttest/interaction_env_logger.go

```diff
@@ -82,17 +82,23 @@ func (l *RedirectLogger) Errorf(format string, v ...interface{}) {
 
 func (l *RedirectLogger) Fatal(v ...interface{}) {
 	l.print(4, v...)
+	panic(v)
```

Comment:
Two comments:
1. Previously it just prints "`FATAL`" , and not to really panicking; so you changed the behavior;
2. `panic` accepts a parameter of type `interface{}`, but the type of v is `[]interface{}`.  You will get the panicking message something like below,
```
panic: ([]interface {}) 0xc00000c030

goroutine 1 [running]:
main.demo(...)
        /.../go/src/examples/varidicTest/main.go:7
main.main()
        /.../go/src/examples/varidicTest/main.go:11 +0x9d
exit status 2
```

It seems `panic(fmt.Sprint(v))` makes more sense than `panic(v)`? 

But I would suggest to  keep the existing behavior? cc @tbg

Reply:
I basically mirrored the `nodeTestHarness` behaviour, see:
https://github.com/etcd-io/raft/blob/c728a6dece7abb918d8fe1c0b397646774028821/node_util_test.go#L61-L79

Point 2 is good, I fixed the panic argument in both loggers.

Re point 1:

It only changes the behaviour of the data-driven tests in `interaction_test.go`, which is the only user of this logger. The way it changes it is a real panic, so if there were tests relying on the old behaviour, it would break them. No tests were broken, so this behaviour wasn't relied upon and this change is ok.

The old behaviour was problematic, because in situations where the code wants to exit with a fatal or panic (e.g. in situations where critical invariants are broken), this logger would instead allow the code to proceed. As a result, there is a high risk that the code will break/fatal/panic in some other place. While preparing this PR, I ran into this exact situation: the code called `logger.Panicf`, proceeded, and ran into a real `panic` elsewhere. Since the log printed into the `logger` is not reported by the datadriven package in such case, I wasn't aware of the first ("root cause") panic, and was looking for a problem elsewhere until eventually I located the earlier panic.

I think we do need to fix this behaviour, but the exact way we do it may vary. The current version of this PR turns it into a real `panic`, which mirrors the other logger in this repo. This seems ok as a quick fix. The longer-term fix should make the output of the `logger` more visible. I reflected it in the TODO in the code.